mesonbuild/backend/backends.py

```diff
@@ -594,6 +614,20 @@ def get_executable_serialisation(
             exe_wrapper = None
 
         workdir = workdir or self.environment.get_build_dir()
+
+        needs_rsp_file = can_use_rsp_file and sum(len(i) for i in cmd_args) >= get_rsp_threshold()
+
+        if needs_rsp_file:
+            hasher = hashlib.sha1()
+            hasher.update(bytes(str(cmd_args), encoding='utf-8'))
```

Comment:
Calling `str` on a list looks strange to me. Maybe could you jalready oin and quote the string here, and use it to compute the hash?

Also, I wasn't aware it was possible to encode a string using the `bytes` constructor. While technically correct, it is more usual to use the `encode` function for doing this.